File: car_factory_app/lib/shared/models/trade.dart
/// PG 없는 직거래 MVP — 웹 `types/trade.ts` 와 동일.
enum DealMethod { meet, transfer }

enum FulfillmentMethod { meetup, delivery }

enum DirectTradeStatus { reserved, completed, cancelled }

class DirectTradeDocument {
  const DirectTradeDocument({
    required this.id,
    required this.orderNumber,
    required this.buyerUid,
    required this.sellerUid,
    required this.productId,
    required this.totalAmount,
    required this.status,
    required this.dealMethod,
    required this.paymentProvider,
  });

  final String id;
  final String orderNumber;
  final String buyerUid;
  final String sellerUid;
  final String productId;
  final int totalAmount;
  final DirectTradeStatus status;
  final DealMethod dealMethod;
  final String paymentProvider;

  factory DirectTradeDocument.fromMap(String id, Map<String, dynamic> map) {
    return DirectTradeDocument(
      id: id,
      orderNumber: map['orderNumber'] as String? ?? '',
      buyerUid: map['buyerUid'] as String? ?? '',
      sellerUid: map['sellerUid'] as String? ?? '',
      productId: map['productId'] as String? ?? '',
      totalAmount: (map['totalAmount'] as num?)?.toInt() ?? 0,
      status: _parseStatus(map['status'] as String?),
      dealMethod: map['dealMethod'] == 'transfer'
          ? DealMethod.transfer
          : DealMethod.meet,
      paymentProvider: map['paymentProvider'] as String? ?? 'direct',
    );
  }

  static DirectTradeStatus _parseStatus(String? value) {
    switch (value) {
      case 'completed':
        return DirectTradeStatus.completed;
      case 'cancelled':
        return DirectTradeStatus.cancelled;
      default:
        return DirectTradeStatus.reserved;
    }
  }
}

File: car_factory/src/types/trade.ts
/** PG 없는 직거래(만나서 결제 / 계좌이체) 주문 */

export type DealMethod = "meet" | "transfer";

/** 판매완료 시점 이행 방식 */
export type FulfillmentMethod = "meetup" | "delivery";

export type DirectTradeStatus = "reserved" | "completed" | "cancelled";

export type TradeTransferInfo = {
  bankName: string;
  accountNumber: string;
  accountHolder: string;
};

export type TradeShippingInfo = {
  carrier: string;
  trackingNumber: string;
};

export type TradeProductSnapshot = {
  title: string;
  thumbnailURL: string;
  price: number;
  partNumber: string | null;
  conditionDescription: string;
};

/** Firestore `orders/{orderId}` — 직거래 MVP (paymentProvider: direct) */
export interface DirectTradeDocument {
  id: string;
  orderNumber: string;
  buyerUid: string;
  buyerDisplayName: string;
  sellerUid: string;
  sellerDisplayName: string;
  productId: string;
  productSnapshot: TradeProductSnapshot;
  quantity: number;
  itemAmount: number;
  shippingFee: number;
  totalAmount: number;
  status: DirectTradeStatus;
  dealMethod: DealMethod;
  /** 판매완료 시 직거래 / 택배 */
  fulfillmentMethod?: FulfillmentMethod | null;
  /** 택배거래 시 운송장 */
  shippingInfo?: TradeShippingInfo | null;
  /** 계좌이체 시 판매자가 등록 */
  transferInfo: TradeTransferInfo | null;
  /** 이후 PG 대비 — 현재는 항상 direct */
  paymentProvider: "direct";
  reservedAt: string;
  completedAt: string | null;
  cancelledAt: string | null;
  cancelledBy: string | null;
  cancelReason: string | null;
  /** 구매자 거래 내역에서 숨김 */
  hiddenForBuyer?: boolean;
  /** 판매자 거래 내역에서 숨김 */
  hiddenForSeller?: boolean;
  createdAt: string;
  updatedAt: string;
}

export type DirectTradeDto = DirectTradeDocument;
